### PR TITLE
docs: improve DataAvailabilityHeader godoc

### PR DESCRIPTION
**Pull request by rootulp**
## Description

Improved the godoc comment for `DataAvailabilityHeader`

### CODE REVIEW

File: pkg/da/data_availability_header.go [hunk 1]
```diff
@@ -18,20 +18,19 @@ const (
...
+// https://arxiv.org/abs/1809.09044 or the Celestia specification:
 // https://github.com/celestiaorg/celestia-specs/blob/master/src/specs/data_structures.md#availabledataheader
 type DataAvailabilityHeader struct {
-	// RowRoot_j 	= root((M_{j,1} || M_{j,2} || ... || M_{j,2k} ))
+	// RowRoot_j = root((M_{j,1} || M_{j,2} || ... || M_{j,2k} ))
 	RowsRoots [][]byte `json:"row_roots"`
 	// ColumnRoot_j = root((M_{1,j} || M_{2,j} || ... || M_{2k,j} ))
 	ColumnRoots [][]byte `json:"column_roots"`
-	// cached result of Hash() not to be recomputed
+	// hash is a cached copy of the Hash() result.
```
Comment (evan-forbes) [inline pkg/da/data_availability_header.go:33]: :thinking: I'm unsure if "copy" is the best word here, since its not a full copy, only a reference

**Review by evan-forbes — COMMENTED** [review]
what's going on with the markdown linter? its doesn't appear to be related to this

**Review by evan-forbes — DISMISSED** [review]

File: pkg/da/data_availability_header.go (hunk 1 above)
Comment (rootulp) [reply pkg/da/data_availability_header.go]: Removed the term "cached copy" in favor of "memoization" in https://github.com/celestiaorg/celestia-app/pull/1078/commits/7efb73198cea95b3dd2558e50632059428e53243

**Comment by rootulp** [comment]
> we have a flakey markdown linter 😭

I updated this PR post https://github.com/celestiaorg/celestia-app/pull/1087 so 🤞

**Comment by rootulp** [comment]
The most recent CI failure https://github.com/celestiaorg/celestia-app/actions/runs/3587022572/jobs/6036912983#step:6:152 should be fixed by https://github.com/celestiaorg/celestia-app/pull/1086

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**